src/main/java/org/wildfly/security/auth/provider/jdbc/mapper/PasswordKeyMapper.java

```diff
@@ -91,33 +91,37 @@
     private final String algorithm;
     private int salt = -1;
     private int iterationCount = -1;
+    private final String credentialName;
     private final Class<?> passwordType;
 
     /**
      * Constructs a new instance.
      *
+     * @param credentialName name of the credential which is output of this mapper
      * @param algorithm the algorithm that will be used by this mapper to create a specific {@link org.wildfly.security.password.Password} type
      * @param hash the column index from where the password in its clear, hash or encoded form is obtained
      * @throws InvalidKeyException if the given algorithm is not supported by this mapper.
      */
-    public PasswordKeyMapper(String algorithm, int hash) throws InvalidKeyException {
+    public PasswordKeyMapper(String credentialName, String algorithm, int hash) throws InvalidKeyException {
```

Comment:
The API for this class is raising more questions at the moment, unfortunately to merge this PR this class would cause changes to be required in the subsystem so we probably need to get this correct first.

@honza889 maybe you can also chat with @pedroigor - @dmlloyd has also had a look and questions if this is largely redundant.